rclcpp/test/rclcpp/test_copy_all_parameters.cpp

```diff
@@ -0,0 +1,57 @@
+// Copyright 2023 Open Navigation LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include "rclcpp/copy_all_parameters.hpp"
+#include "rclcpp/rclcpp.hpp"
+
+class RclCppFixture
+{
+public:
+  RclCppFixture() {rclcpp::init(0, nullptr);}
+  ~RclCppFixture() {rclcpp::shutdown();}
+};
+RclCppFixture g_rclcppfixture;
```

Comment:
Is this a pattern copied from somewhere else?

The only pattern I'm familiar with is ones like this:

https://github.com/ros2/rclcpp/blob/ea31f94eb453acf3f34b37b85e38284abc514d57/rclcpp/test/rclcpp/test_node.cpp#L41-L65

Using `TEST_F`.

If we keep the current code, then I think (according to our style) it should be named `RclcppFixture` or maybe `RCLCPPFixture`, but `RclCppFixture` stands out as strange in the code base I think.

Reply:
Its a pattern we use in Nav2 tests alot. I can't speak to the commonality of it, I've been doing it for so long I don't remember its origin. It works well as a global to setup the init when the test starts and shut it down once all the tests in a file end. I think its nicer than the `TestNode` example, but its not my sandbox. 

I don't feel strongly about it either way. I just updated the naming though to be in line. Happy to adjust should it be requested.

Reply:
I don't have a problem with it for any strong reason, only that I could imagine it might not play well with gtest if you were to, for example, have it loop tests or something. It seems like a good idea to use the gtest mechanism for setup/teardown if we can.

Why is it nicer, in your opinion?

Reply:
Also, it would be more consistent with the other tests. So I would say it would be better to use the gtest style.

Either way, the name should be changed.

Reply:
It should work for that fine. This simply sets up init at the start of the file's execution and takes it down once the file is completed. If you were to be looping it, each of those executions are separate. 

> Why is it nicer, in your opinion?

Look at the amount of code :smile: Its like a little standard preamble (this is actually automatically generated via my `maketest` script) 

OK, I can change it, no problem.